From 9a1b1ad5342f21c3f58501cd7d2e9b824746ca8b Mon Sep 17 00:00:00 2001
From: Dwane Pottratz <dwane.pottratz@intel.com>
Date: Tue, 10 Nov 2015 14:25:06 -0800
Subject: [PATCH 15/17] input: touch: silead: Emdoor i8880 broken multi-touch

the i8880 with the di_xian panel firmware is badly broken for multi touch
We test for the firmware version and don't use the data slot information
but instead use the touch number

Change-Id: Iefcd840021c7bfe7250dbbb15c0349ec43cf0779
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-20919
Signed-off-by: Dwane Pottratz <dwane.pottratz@intel.com>
Signed-off-by: Richard A. Griffiths <richard.a.griffiths@intel.com>
Reviewed-on: https://android.intel.com:443/472788
---
 drivers/input/touchscreen/silead.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/drivers/input/touchscreen/silead.c b/drivers/input/touchscreen/silead.c
index c4b225a..4b334bf 100644
--- a/drivers/input/touchscreen/silead.c
+++ b/drivers/input/touchscreen/silead.c
@@ -208,7 +208,7 @@ static void silead_ts_read_data(struct i2c_client *client)
 		data->pos[i - 1].x = x;
 		data->pos[i - 1].y = y;
 
-		dev_dbg(dev, "x=%d y=%d id=%d\n", x, y, id);
+		dev_dbg(dev, "x=%d y=%d id=%d, fw_name=%s\n", x, y, id, data->fw_name);
 	}
 
 	input_mt_assign_slots(data->input_dev, data->slots, data->pos, touch_nr);
@@ -217,6 +217,14 @@ static void silead_ts_read_data(struct i2c_client *client)
 		x = data->pos[i].x;
 		y = data->pos[i].y;
 		id = data->slots[i];
+/* the i8880 with the di_xian panel firmware is badly broken for multi touch
+ * We test for the firmware version and don't use the data slot information
+ * but instead use the touch number
+ */
+		if(strstr(data->fw_name,"di_xian") != NULL){
+			dev_dbg(dev, "%s in use no data slot support. using touch_nr\n", data->fw_name);
+			id = i;
+		}
 
 		if (data->xy_swap)
 			silead_ts_report_touch(data,
-- 
1.9.1

